### PR TITLE
299 create bridge documentation

### PR DESCRIPTION
**Pull request by leoloco**

### CODE REVIEW

File: docs/massaBridge/home.mdx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+## Supported Wallets
+
+For Ethereum:
+- [Metamask](https://metamask.io/) wallet
+
+For Massa:
+- [Massa Station’s Wallet module](https://station.massa.net/) (recommended by Massa Labs)
+- [Bearby wallet](https://bearby.io/) (community-made wallet)
+
+For more details, please refer to the Massa Bridge FAQ: [https://bridge.massa.net/](https://bridge.massa.net/).
```
Comment (longgamma-massa) [inline docs/massaBridge/home.mdx:29]: Add link to FAQ: https://bridge.massa.net/faq

File: docs/massaBridge/instructions.mdx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+This page explains step-by-step instructions on how to bridge tokens to and from Massa networks.
+
+## Connect Wallets
+
+1. Navigate to [Massa Bridge](https://bridge.massa.net/index). Make sure the mode is set to ‘Mainnet’. To try the testnet Bridge, switch the mode to Testnet in the top-right of your screen.
+2. Click on the ‘Connect wallet’ button.
+3. Connect your Metamask wallet on the Ethereum network.
+4. Connect a Massa wallet account:
+   - For Massa Station Wallet (recommended): Ensure the Massa Station desktop app is installed, open, and running. For Massa Station-related questions, please refer to [Massa Station FAQ](https://docs.massa.net/docs/massaStation/faq) and [Massa Wallet FAQ](https://docs.massa.net/docs/massaStation/massa-wallet/getting-started).
```
Comment (longgamma-massa) [inline docs/massaBridge/instructions.mdx:11]:
```suggestion
   - For Massa Station Wallet (recommended): Ensure the Massa Station desktop app is installed, open, and running. For Massa Station-related questions, please refer to [Massa Station FAQ](https://docs.massa.net/docs/massaStation/faq) and [Massa Wallet Instructions](https://docs.massa.net/docs/massaStation/massa-wallet/getting-started).
```suggestion
   - For Massa Station Wallet (recommended): Ensure the Massa Station desktop app is installed, open, and running. For Massa Station-related questions, please refer to [Massa Station FAQ](https://docs.massa.net/docs/massaStation/faq) and [Massa Wallet FAQ](https://docs.massa.net/docs/massaStation/massa-wallet/getting-started).
```
```

File: docs/massaBridge/instructions.mdx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+   - For Bearby (community-developed): Ensure the extension is installed and activated.
+
+## Bridge Tokens from Ethereum to Massa
+
+Note: You will need to sign up to 2 transactions on Ethereum, and none on Massa.
```
Comment (longgamma-massa) [inline docs/massaBridge/instructions.mdx:16]:
```suggestion
Note: You will need to sign up to 2 transactions on Ethereum, and none on Massa.
```
:::note
You will need to sign up to 2 transactions on Ethereum, and none on Massa. 
:::

File: docs/massaBridge/instructions.mdx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+4. If bridging tokens for the first time, you must first sign the Allowance transaction. In the Metamask’s sign request screen, you can approve the max. or a custom amount, depending on your preference. Proceed to sign the transaction.
+5. Once the allowance transaction is completed, you will be asked to sign the bridge transaction in your Metamask. Review the transaction and sign it.
+6. Return to the Bridge page and wait for the completion of your transfer.
+7. A success screen will be shown once the Massa operation is finalized. The tokens should now be in your wallet.
+8. Follow the instructions to add the token to your Massa wallet.
+9. Past transfers can be checked on the History page.
+
+## Redeem Tokens from Massa to Ethereum
+
+Note: You will need to sign up to 3 transactions: 2 on Massa, and 1 on Ethereum network.
```
Comment (longgamma-massa) [inline docs/massaBridge/instructions.mdx:30]:
```suggestion
Note: You will need to sign up to 3 transactions: 2 on Massa, and 1 on Ethereum network.
```
:::note
You will need to sign up to 3 transactions: 2 on Massa, and 1 on Ethereum network.
:::

File: docs/massaBridge/architecture-security.mdx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+### Security Measures
+
+The bridgeVault contract on Ethereum holds the assets locked during the bridging process. The only way assets can be redeemed is by sending a message signed by the 3 out of 3 nodes of the relayer network. This ensures that the assets are only released when the relayer network has verified the corresponding burn transaction on Massa. The public keys of the relayer nodes are stored in the bridge contract on Massa, and can only be updated by a gnosis safe multisig wallet controlled by the bridge operators. Moreover, a 48h timeLock from openzeppelin is implemented in the bridge contract on Massa, which requires a 48h delay before the relayer network can update the public keys. This ensures that the bridge operators have enough time to intervene in case of a malicious update of the public keys.
```
Comment (longgamma-massa) [inline docs/massaBridge/architecture-security.mdx:25]: you address this contract as 'BridgeVault' in the above content, so let's use the same naming below as well for clarity.

File: docs/massaBridge/architecture-security.mdx [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+##### EVM blockchain interaction:
+
+- Once sufficient **Relayer network** signatures are collected, the **Bridge website** prepares the corresponding transaction.
+- The **User** signs and sends the redeem transaction to the **EVM blockchain** using their wallet provider.
+- Once the redeem transaction is executed on the **EVM blockchain**, the **User** is notified of its completion.
+
+
+### Off-chain Relayer Network
+![Off chain architecture](off-chain-architecture.png)
```
Comment (longgamma-massa) [inline docs/massaBridge/architecture-security.mdx:87]: I wonder if sharing this info in so much detail is smart. For example: to a normal user who wants to ensure himself the bridge is secure, knowing that we use: EC2, lambda, amplify etc doesn't mean much. For someone wanting to exploit Bridge, it is valuable info (correct me if I am wrong). If you agree, I suggest that we simplify this scheme by mentioning vaguely backend architecture without these specifics.

File: docs/massaBridge/architecture-security.mdx (hunk 5 above)
Comment (longgamma-massa) [reply docs/massaBridge/architecture-security.mdx:25]: OpenZeppelin instead openzeppelin

**Review by longgamma-massa — CHANGES_REQUESTED** [review]
@leoloco for the preference of writing: I prefer yours over gpt. 
Left a few comments and suggestions

File: docs/massaBridge/architecture-security.mdx [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+### Security Measures
+
+The bridgeVault contract on Ethereum holds the assets locked during the bridging process. The only way assets can be redeemed is by sending a message signed by the 3 out of 3 nodes of the relayer network. This ensures that the assets are only released when the relayer network has verified the corresponding burn transaction on Massa. The public keys of the relayer nodes are stored in the bridge contract on Massa, and can only be updated by a gnosis safe multisig wallet controlled by the bridge operators. Moreover, a 48h timeLock from openzeppelin is implemented in the bridge contract on Massa, which requires a 48h delay before the relayer network can update the public keys. This ensures that the bridge operators have enough time to intervene in case of a malicious update of the public keys. 
+The bridgeVault contract can charge fees. However they have been set to 0 for the time being.
+
+**Gnosis safe**
+Safe address: [0x1234567890abcdef1234567890abcdef12345678](https://gnosis-safe.io/app/#/safes/0x1234567890abcdef1234567890abcdef12345678)
+
```
[the remark this answers is not in the document]
Comment (leoloco) [reply docs/massaBridge/architecture-security.mdx:30]: From certik audit page 16: `A medium/blog link for sharing the timelock contract and multi-signers addresses information with the public audience.`

File: docs/massaBridge/architecture-security.mdx (hunk 6 above)
Comment (leoloco) [reply docs/massaBridge/architecture-security.mdx:87]: Okay, so the previous diagram is sufficient, let's just delete this diagram.

Comment (leoloco) [reply docs/massaBridge/architecture-security.mdx:87]: FYI, reverse engineering our relayer network would be a matter of minutes for any serious hacker. Obfuscating our architecture won't impact our security. But being too vague might not be reassuring for legit users

**Comment by longgamma-massa** [comment]
@peterjah can you review architecture and security page?

**Comment by longgamma-massa** [comment]
@SlnPons check overall documentation  
@gregLibert can you review Architecture and Security page so that Leo has a reviewer?

File: docs/massaBridge/architecture-security.mdx [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+The BridgeVault contract can charge fees. However they have been set to 0 for the time being.
+
+**Gnosis safe**
+Safe proxy address: [0x5d752840501b2d2f4841accf53794d0ed3e2c3eb](https://sepolia.etherscan.io/address/0x5d752840501b2d2f4841accf53794d0ed3e2c3eb)
+
+| Signer Name | Address                                   | Status   |
+|-------------|-------------------------------------------|----------|
+| Damir       | 0xAbC1234...EfgH5678                      | Active   |
+| Adrien      | 0xIjk9012...LmnO3456                      | Active   |
+| SRE         | 0xPqr7890...StuV1234                      | Inactive |
```
Comment (gregLibert) [inline docs/massaBridge/architecture-security.mdx:37]: Why is it inactive ?

File: docs/massaBridge/architecture-security.mdx [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+![Bridge flow](bridge-flow.png)
+
+### Security Measures
+
+The BridgeVault contract on Ethereum holds the assets locked during the bridging process. The only way assets can be redeemed is by sending a message signed by the 3 out of 3 nodes of the relayer network. This ensures that the assets are only released when the relayer network has verified the corresponding burn transaction on Massa. The public keys of the relayer nodes are stored in the Bridge contract on Massa, and can only be updated by a gnosis safe multisig wallet controlled by the bridge operators. Moreover, a 48h timeLock from Openzeppelin is implemented in the Bridge contract on Massa, which requires a 48h delay before the relayer network can update the public keys. This ensures that the bridge operators have enough time to intervene in case of a malicious update of the public keys. 
+The BridgeVault contract can charge fees. However they have been set to 0 for the time being.
+
+**Gnosis safe**
+Safe proxy address: [0x5d752840501b2d2f4841accf53794d0ed3e2c3eb](https://sepolia.etherscan.io/address/0x5d752840501b2d2f4841accf53794d0ed3e2c3eb)
```
Comment (gregLibert) [inline docs/massaBridge/architecture-security.mdx:31]: What is it on buildnet?

File: docs/massaBridge/architecture-security.mdx [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+
+The Bridge contract on Massa is responsible for minting the tokens in response to the locked assets on Ethereum. The Bridge contract can only mint tokens when it receives a message signed by 3 out of 3 nodes of the relayer network. The synthetic token contracts ownership, has yet to be transferred to a multisig wallet after deployment. The public keys of the relayer nodes are hardcoded in the contract and cannot be updated. This contract as well can be paused for maintenance or security.
+The Bridge contract can charge fees but have been set to 0.
+
+![On chain architecture](on-chain-architecture.png)
+
+The relayer network is a set of independent nodes responsible for monitoring bridge transactions, verifying events, and facilitating the cross-chain communication necessary for asset transfers. The three nodes are hosted on different cloud providers to avoid common points of failure. For redundancy, each node has a fallback mechanism and redundant RPC connections.
+
+The Massa Bridge smart contracts have undergone security audits conducted by Certik, the report is publicly available [here](https://drive.google.com/file/d/1onla7nqsow26RZYXfJE2ju0vGwdjfttA/view?usp=sharing).
```
Comment (gregLibert) [inline docs/massaBridge/architecture-security.mdx:47]: Certifk doesn't provide a public link?

File: docs/massaBridge/home.mdx [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+## Token Addresses
+
+| Network | Token name | Address |
+|---------|-------|---------|
+| Ethereum | USDC | _add token contract link_ |
+| Ethereum | WETH | _add token contract link_ |
+| Ethereum | DAI | _add token contract link_ |
+| Massa | USDC.e | _add token contract link_ |
+| Massa | WETH.e | _add token contract link_ |
+| Massa | DAI.e | _add token contract link_ |
```
Comment (gregLibert) [inline docs/massaBridge/home.mdx:18]: to be updated?

File: docs/massaBridge/architecture-security.mdx (hunk 10 above)
Comment (gregLibert) [reply docs/massaBridge/architecture-security.mdx:47]: Publishing is in progress.

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: docs/massaBridge/home.mdx (hunk 11 above)
Comment (leoloco) [reply docs/massaBridge/home.mdx:18]: yes indeed

**Review by gregLibert — APPROVED** [review]

**Review by gregLibert — APPROVED** [review]

**Review by longgamma-massa — APPROVED** [review]

**Pull request merged**